// math.proto
syntax = "proto3";

package math;

service MathService {
  rpc IsPerfectNumber (NumberRequest) returns (BoolResponse) {}
  rpc Gcd (GcdRequest) returns (GcdResponse) {}
}

message NumberRequest {
  int64 number = 1;
}

message BoolResponse {
  bool is_perfect = 1;
}

message GcdRequest {
  int64 a = 1;
  int64 b = 2;
}

message GcdResponse {
  int64 gcd = 1;
}
